src/groovy/com/unifina/signalpath/messaging/EmailModule.groovy

```diff
@@ -59,9 +58,11 @@ class EmailModule extends ModuleWithSideEffects {
 		parentSignalPath?.showNotification(getMessageBody())
 	}
 
-	private String getMessageBody() {
-		initDf()
+	private String linkToCanvas() {
+		return String.format("https://www.streamr.com/canvas/editor/%s", parentSignalPath.canvas.id)
```

Comment:
- Should not hard-code URL here. Maybe abstract the url creation to `canvasService` (something like `getCanvasURL(Canvas canvas)` and there use the [grailsLinkGenerator](http://mrhaki.blogspot.com/2012/01/grails-goodness-generate-links-outside.html) bean to generate the link (remember to set `absolute: true`). Then in the Email module use `canvasService` to get the URL.

- Use `getRootSignalPath().canvas` to generate the link instead of `parentSignalPath.canvas`, as `canvas` will probably be null for "sub-SignalPaths"